src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,490 @@
+                                tcsConnect.SetResult();
+                            }
+                        }
+                        else
+                        {
+                            tcsConnect.SetCanceled(cancellationToken);
+                        }
+                    }
+                });
+
+                // Attach the onOpen callback
+                _innerWebSocket.SetObjectProperty("onopen", _onOpen);
+
+                // Setup the onMessage callback
+                _onMessage = new Action<JSObject>((messageEvent) =>
+                {
+                    // get the events "data"
+                    using (messageEvent)
+                    {
+                        ThrowIfNotConnected();
+                        // If the messageEvent's data property is marshalled as a JSObject then we are dealing with
+                        // binary data
+                        object eventData = messageEvent.GetObjectProperty("data");
+                        switch (eventData)
+                        {
+                            case ArrayBuffer buffer:
+                                using (buffer)
+                                {
+                                    _receiveMessageQueue.Writer.TryWrite(new ReceivePayload(buffer, WebSocketMessageType.Binary));
+                                    break;
+                                }
+                            case JSObject blobData:
+                                using (blobData)
+                                {
+                                    Action<JSObject>? loadend = null;
+                                    // Create a new "FileReader" object
+                                    using (HostObject reader = new HostObject("FileReader"))
+                                    {
+                                        loadend = (loadEvent) =>
+                                        {
+                                            using (loadEvent)
+                                            using (JSObject target = (JSObject)loadEvent.GetObjectProperty("target"))
+                                            {
+                                                // https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readyState
+                                                if ((int)target.GetObjectProperty("readyState") == 2) // DONE - The operation is complete.
+                                                {
+                                                    using (ArrayBuffer binResult = (ArrayBuffer)target.GetObjectProperty("result"))
+                                                    {
+                                                        _receiveMessageQueue.Writer.TryWrite(new ReceivePayload(binResult, WebSocketMessageType.Binary));
+                                                        if (loadend != null)
+                                                            JavaScript.Runtime.FreeObject(loadend);
```

Comment:
I don't understand this part, which object are you trying to release?

Reply:
Cleaned up deprecated pre ref counting code.